server/routes/nl/helpers.py

```diff
@@ -62,19 +61,17 @@
 
 # Get the default place to be used for fulfillment. If there is a place in the
 # request, use that. Otherwise, use pre-chosen places.
-def _get_default_place(request: Dict, is_sdg: bool):
+def _get_default_place(request: Dict, is_sdg: bool, debug_logs: Dict):
   default_place_dcid = request.args.get('default_place', default='', type=str)
   # If default place from request is earth, use the Earth place object
   if default_place_dcid == constants.EARTH.dcid:
     return constants.EARTH
-  # If default place from request is something else, create a place object
+  # If default place from request is something else, get the place object for
+  # that dcid
   elif default_place_dcid:
-    if default_place_dcid.startswith(constants.COUNTRY_DCID_PREFIX):
-      place_type = 'Country'
-    else:
-      place_type = get_place_type(default_place_dcid)
-    place_name = names([default_place_dcid])[default_place_dcid]
-    return Place(default_place_dcid, place_name, place_type)
+    places, _ = get_place_from_dcids([default_place_dcid], debug_logs)
+    if len(places) > 0:
+      return places[0]
```

Comment:
If for whatever reason places is empty, then this function won't return anything?  Perhaps we want to return None and check in the caller?

Reply:
Actually, what should the caller do if there is no default place returned? Should the caller just continue without a default place or use the original logic for getting default place?

Reply:
Hmmm, three options:
1. Use the original logic
2. Make up a place as:  `Place(type='Place', name=dcid, dcid=dcid)`
3. Continue without default place (aka return None here)

I prefer (1) less.  (2) assumes the DCID exists, which may fail in other ways.

So I was suggesting (3), in which case, I think we would return an error message like no place found in the query.  Perhaps test it out...

Reply:
updated to return an error message when there is no place: https://screenshot.googleplex.com/587DDdCAGTF2XH2